gems/aws-sdk-s3/lib/aws-sdk-s3/customizations/object.rb

```diff
@@ -496,8 +496,9 @@ def upload_file(source, options = {})
       #   the object has a stored checksum, it will be used to validate the
       #   download and will raise an `Aws::Errors::ChecksumError` if
       #   checksum validation fails. You may provide a `on_checksum_validated`
-      #   callback if you need to verify that validation occured and which
-      #   algorithm was used.
+      #   callback if you need to verify that validation occurred and which
+      #   algorithm was used.  To disable checksum validation, set
+      #   `checksum_mode` to either `false` or `nil`.
```

Comment:
This option is a string. Should we actually take 'DISABLED' here?

Reply:
DISABLED may eventually be an api enum option and we'd want to pass that through.  Right now sending DISABLED will cause an error.

Reply:
Updated: support setting to "DISABLED"